src/lib/swapper/swappers/ArbitrumBridgeSwapper/utils/fetchArbitrumBridgeSwap.ts

```diff
@@ -0,0 +1,63 @@
+import { Erc20Bridger, EthBridger, getL2Network } from '@arbitrum/sdk'
+import type {
+  L1ToL2TransactionRequest,
+  L2ToL1TransactionRequest,
+} from '@arbitrum/sdk/dist/lib/dataEntities/transactionRequest'
+import { ethAssetId, ethChainId, fromAssetId } from '@shapeshiftoss/caip'
+import { isEvmChainId } from '@shapeshiftoss/chain-adapters'
+import { type Asset, KnownChainIds } from '@shapeshiftoss/types'
+import { BigNumber } from 'ethers5'
+import { arbitrum } from 'viem/chains'
+import { getEthersV5Provider } from 'lib/ethersProviderSingleton'
+
+export type FetchArbitrumBridgeSwapInput = {
+  affiliateBps: string
+  buyAsset: Asset
+  receiveAddress: string
+  sellAmountIncludingProtocolFeesCryptoBaseUnit: string
+  sellAsset: Asset
+  maximumSlippageDecimalPercentage: string
+  sendAddress: string
+}
+
+export const fetchArbitrumBridgeSwap = async ({
+  buyAsset,
+  sellAmountIncludingProtocolFeesCryptoBaseUnit,
+  sellAsset,
+  sendAddress,
+  receiveAddress,
+}: FetchArbitrumBridgeSwapInput): Promise<
+  Omit<L1ToL2TransactionRequest | L2ToL1TransactionRequest, 'retryableData'>
+> => {
+  const l2Network = await getL2Network(arbitrum.id)
+  const isDeposit = sellAsset.chainId === ethChainId
+  const isEthBridge = isDeposit ? sellAsset.assetId === ethAssetId : buyAsset.assetId === ethAssetId
+
+  const bridger = isEthBridge ? new EthBridger(l2Network) : new Erc20Bridger(l2Network)
+
+  if (!(isEvmChainId(sellAsset.chainId) && isEvmChainId(buyAsset.chainId))) {
+    throw new Error(`Arbitrum Bridge only supports EVM chains`)
+  }
+
+  const erc20L1Address = fromAssetId((isDeposit ? sellAsset : buyAsset).assetId).assetReference
```

Comment:
I would like to see eth bridge separated from token bridge in this context for clarity. It is mixed and while the appropriate params will be used for each bridge it would be nice to see the two bridge interactions and what values they care about imo

Reply:
531599870998817f1ba55ea7ad7be8412ad5db02 - TBD in getTradeQuote too which is a bigger and currently more spaghettiy boi

Reply:
Also https://github.com/shapeshift/web/pull/6977/commits/ae57872de7371812330b00aebdddcf300aeb4f94  for follow-up cleanup